modal/client.py

```diff
@@ -412,6 +405,22 @@ def name(self) -> str:
         return self.wrapped_method.name
 
     async def __call__(
+        self,
+        req: RequestType,
+        retry: Optional[Union[Retry, NoRetry]] = None,
```

Comment:
I think I'm still not understanding the difference between `retry=NoRetry()` and `retry=None`, why do you need both? Also was there a reason you decided not to go with having an instance default?